engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/VectorTest.java

```diff
@@ -139,6 +142,90 @@ public void passingListOrArrayToEnsoAsArray() throws Exception {
     assertEquals("Hundred elements", 100, hundred.asInt());
   }
 
+  @Test
+  public void vectorWithWarningViaForEach() throws Exception {
+    warningsInContainer(1, 1);
```

Comment:
Use `testOnly *VectorTest` to debug behavior of warnings in an _isolation_ (from the `Standard.Test` harness):

- warnings behavior depends on every detail
- there are deeper nuances elsewhere,
- thinking about warnings from Enso code macroscopic perspective is hard,
   - it is "black box testing" (without clear specification),
- one needs to **debug the interpreter** to really know what's going on